projects/plugins/super-cache/inc/preload-notification.js

```diff
@@ -1,18 +1,92 @@
 jQuery( document ).ready( function () {
-	load_preload_status();
-	setInterval( function () {
-		load_preload_status();
-	}, 3000 );
-} );
-
-/**
- *
- */
-function load_preload_status() {
-	jQuery.get( {
-		url: wpsc_preload_ajax.preload_permalink_url + '?' + Math.random(),
-		success: function ( response ) {
-			jQuery( '#preload_status' ).text( response );
-		},
-	} );
-}
+	const { __, sprintf } = wp.i18n;
+
+	// Set the maximum number of entries to show
+	const MAX_PRELOAD_ENTRIES = 5;
+
+	// Set how often to check.
+	const CHECK_INTERVAL = 3000;
+
+	// Get a reference to the log element and the previous log entry
+	const preloadInfoPanel = jQuery( "#wpsc-preload-status" );
+
+	// Abort early if no info panel exists.
+	if ( ! preloadInfoPanel.length ) {
+		return;
+	}
+
+	update_preload_status( wpsc_preload_ajax.preload_status );
+
+	setInterval( () => {
+		jQuery.post(
+			wpsc_preload_ajax.ajax_url,
+			{
+				action: 'wpsc_get_preload_status',
+			},
+			( json ) => {
+				if ( ! json || ! json.success ) {
+					return;
+				}
+
+				update_preload_status( json.data );
+			}
+		);
+	}, CHECK_INTERVAL );
```

Comment:
I'm not a huge fan of relying on the user's connection being fast enough to finish the request in the time specified in `CHECK_INTERVAL`. I know it's an edge case, but it makes my senses tingle :D

Reply:
It's 3 seconds, and it used to be 1 second. I don't think it's much to worry about though as it's only for the notification. The AJAX request is fairly light weight too.

Reply:
🤔 Defensive code is a good idea. In 4d41ccdc4f I made it use `setTimeout` which is called after each request completes so each will be 3 seconds after the last is completed.

It will also stop if one of the update requests fails, so that we don't keep polling a broken server. If users need to resume after an error they can refresh the page.